src/main/java/com/rarchives/ripme/ripper/rippers/E621Ripper.java

```diff
@@ -25,118 +27,141 @@
  * @author
  */
 public class E621Ripper extends AbstractHTMLRipper{
-	private static Pattern gidPattern=null;
-	private static Pattern gidPattern2=null;
-	private static Pattern gidPatternPool=null;
+public static final int POOL_IMAGES_PER_PAGE = 24;
 	
-	private DownloadThreadPool e621ThreadPool=new DownloadThreadPool("e621");
+	private DownloadThreadPool e621ThreadPool = new DownloadThreadPool("e621");
 	
 	public E621Ripper(URL url) throws IOException {
 		super(url);
 	}
-
+	
 	@Override
 	public DownloadThreadPool getThreadPool() {
 		return e621ThreadPool;
 	}
-
+	
 	@Override
 	public String getDomain() {
 		return "e621.net";
 	}
-
+	
 	@Override
 	public String getHost() {
 		return "e621";
 	}
-
+	
 	@Override
 	public Document getFirstPage() throws IOException {
-		if(url.getPath().startsWith("/pool/show/"))
-			return Http.url("https://e621.net/pool/show/"+getTerm(url)).get();
+		if (url.getPath().startsWith("/pool/show/"))
+			return Http.url("https://e621.net/pool/show/" + getTerm(url)).get();
 		else
-			return Http.url("https://e621.net/post/index/1/"+getTerm(url)).get();
+			return Http.url("https://e621.net/post/index/1/" + getTerm(url)).get();
 	}
-
+	
 	@Override
 	public List<String> getURLsFromPage(Document page) {
-		Elements elements=page.select("#post-list .thumb a,#pool-show .thumb a");
-		List<String> res=new ArrayList<String>(elements.size());
+		Elements elements = page.select("#post-list .thumb a,#pool-show .thumb a");
+		List<String> res = new ArrayList<String>(elements.size());
 		
-		for(Element e:elements){
-			res.add(e.absUrl("href")+"#"+e.child(0).attr("id").substring(1));
+		if (page.getElementById("pool-show") != null) {
+			int index = 0;
+			
+			Element e = page.getElementById("paginator");
+			if (e != null && (e = e.getElementsByClass("current").first()) != null)
```

Comment:
Please don't use assignment within an if-statement, as it makes reading and maintaining the code more difficult. Nested if-statements are fine. Please also wrap if-statement and for-loop bodies in braces.